pkg/pushsync/pushsync.go

```diff
@@ -224,9 +222,9 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk) (rr *pb.R
 		deferFuncs = append(deferFuncs, func() { go streamer.FullClose() })
 
 		w, r := protobuf.NewWriterAndReader(streamer)
-		ctx, cancel := context.WithTimeout(ctx, timeToWaitForReceipt)
-		defer cancel()
-		if err := w.WriteMsgWithContext(ctx, &pb.Delivery{
+		ctxd, canceld := context.WithTimeout(ctx, timeToLive)
```

Comment:
This context `ctxd` has a timeout of `timeToLive` which is the same as the inherited context created in the handler function. Basically, this timeout it is not needed if they are of the same value. The same goes for `ctxr`. Either to raise the `ctx` timeout or to just pass `ctx` directly to `WriteMsgWithContext` and `WriteMsgWithContext` functions without creating additional timeouts here.

Reply:
actually, one of them is needed but the other isn't. The problem is that this same method is shared both by `pusher` and `handler`. If the deadline is already set by handler then the context will expire before the new deadline is reached. I think we can live with this, at least until we refactor the package to wait for responses from already open streams instead of just closing and trying a different one